File: backend/profile-service/src/main/java/com/example/profileservice/profile/repository/ProfileCustomRepository.java
package com.example.profileservice.profile.repository;

import com.example.profileservice.profile.entity.Profile;

import java.util.List;

public interface ProfileCustomRepository {
    Profile findByMemberProfile(Long memberId, Long id);

    Profile findLatestProfile(Long memberId);

    List<Profile> findByMemberProfileOffset(Long memberId, Long id, Long count);

    Long findMemberTotalProfiles(Long memberId);

    Long deleteByMemberProfile(Long memberId, Long id);
}

File: backend/profile-service/src/main/kotlin/com/example/profileservice/profile/repository/ProfileCustomRepository.kt
package com.example.profileservice.profile.repository

import com.example.profileservice.profile.entity.Profile

interface ProfileCustomRepository {
    fun findByMemberProfile(memberId: Long, id: Long): Profile?

    fun findLatestProfile(memberId: Long): Profile?

    fun findByMemberProfileOffset(memberId: Long, id: Long, count: Long): List<Profile>

    fun findMemberTotalProfiles(memberId: Long): Long

    fun deleteByMemberProfile(memberId: Long, id: Long): Long
}
